pydis_site/apps/api/tests/test_reminders.py

```diff
@@ -0,0 +1,196 @@
+            'author': self.author.id,  # Missing multiple required fields
+        }
+        url = reverse('bot:reminder-list', host='api')
+        response = self.client.post(url, data=data)
+        self.assertEqual(response.status_code, 400)
+        self.assertRaises(Reminder.DoesNotExist, Reminder.objects.get, id=1)
+
+
+class ReminderDeletionTests(APISubdomainTestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.author = User.objects.create(
+            id=6789,
+            name='Barnacle Boy',
+            discriminator=6789,
+            avatar_hash=None,
+        )
+
+        cls.reminder = Reminder.objects.create(
+            author=cls.author,
+            content="Don't forget to set yourself a reminder",
+            expiration=datetime.utcnow().isoformat(),
+            jump_url="https://www.decliningmentalfaculties.com",
+            channel_id=123
+        )
+
+    def test_delete_unknown_reminder_returns_404(self):
+        url = reverse('bot:reminder-detail', args=('something',), host='api')
+        response = self.client.delete(url)
+
+        self.assertEqual(response.status_code, 404)
+
+    def test_delete_known_reminder_returns_204(self):
+        url = reverse('bot:reminder-detail', args=(self.reminder.id,), host='api')
+        response = self.client.delete(url)
+
+        self.assertEqual(response.status_code, 204)
+        self.assertRaises(Reminder.DoesNotExist, Reminder.objects.get, id=self.reminder.id)
+
+
+class ReminderListTests(APISubdomainTestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.author = User.objects.create(
+            id=6789,
+            name='Patrick Star',
+            discriminator=6789,
+            avatar_hash=None,
+        )
+
+        cls.reminder_one = Reminder.objects.create(
+            author=cls.author,
+            content="We should take Bikini Bottom, and push it somewhere else!",
+            expiration=datetime.utcnow().isoformat(),
+            jump_url="https://www.icantseemyforehead.com",
+            channel_id=123
+        )
+
+        cls.reminder_two = Reminder.objects.create(
+            author=cls.author,
+            content="Gahhh-I love being purple!",
+            expiration=datetime.utcnow().isoformat(),
+            jump_url="https://www.goofygoobersicecreampartyboat.com",
+            channel_id=123,
+            active=False
+        )
+
+        cls.rem_dict_one = model_to_dict(cls.reminder_one)
+        cls.rem_dict_one['expiration'] += 'Z'  # Massaging a quirk of the response time format
+        cls.rem_dict_two = model_to_dict(cls.reminder_two)
+        cls.rem_dict_two['expiration'] += 'Z'  # Massaging a quirk of the response time format
+
+    def test_reminders_in_full_list(self):
+        url = reverse('bot:reminder-list', host='api')
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), [self.rem_dict_one, self.rem_dict_two])
```

Comment:
I think this test is a bit fragile. Formally, the model has no default `ordering` defined, which makes the order in which the objects are returned "unspecified":

> If a query doesn’t have an ordering specified, results are returned from the database in an unspecified order. A particular ordering is guaranteed only when ordering by a set of fields that uniquely identify each object in the results. For example, if a name field isn’t unique, ordering by it won’t guarantee objects with the same name always appear in the same order. ([source](https://docs.djangoproject.com/en/3.0/ref/models/options/#ordering))

I'm not sure what the best approach is to making these tests robust against ordering issues, as I don't think they're sortable by default. (Otherwise we could sort both lists and compare them after a sort.)

A way around that could be to use [`TestCase.assertCountEqual`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertCountEqual) which tests if the lists contain the same elements with the same counts regardless of order.

```suggestion
        self.assertCountEqual(response.json(), [self.rem_dict_one, self.rem_dict_two])
```

Reply:
Whoops, I've never really thought too much about list equality comparison in Python. Considering how convenient everything else tends to be, I just assumed it'd be dealt with automagically. `TestCase.assertCountEqual` seems like a good solution!